.github/workflows/linkcheck.yml

```
@@ -1,43 +1,59 @@
+# Reference:
+#   - https://github.com/actions/cache
+#   - https://github.com/actions/checkout
+#   - https://github.com/lycheeverse/lychee-action
+#   - https://github.com/peter-evans/create-issue-from-file
+
 name: LinkCheck - All files 🔗
 
 on:
   schedule:
     - cron: "0 5 * * 0" # Runs at 05:00 on Sunday.
   workflow_dispatch:
+
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
-permissions:
-  id-token: none
+defaults:
+  run:
+    shell: bash -l {0}
 
 jobs:
-  doc:
-    name: Check Links
-    runs-on: ubuntu-22.04
-    env:
-      PYVISTA_OFF_SCREEN: "True"
-      ALLOW_PLOTTING: true
-      SHELLOPTS: "errexit:pipefail"
+  linkcheck:
+    runs-on: "ubuntu-22.04"
+
+    permissions:
+      issues: write # required for peter-evans/create-issue-from-file
+
     steps:
       - uses: actions/checkout@v4
         with:
+          fetch-depth: 0
           persist-credentials: false
 
-      - uses: actions/setup-python@v5
+      - name: "Cache Lychee"
+        uses: actions/cache@v4
         with:
-          python-version: "3.13"
-          cache: "pip"
+          path: .lycheecache
+          key: cache-lychee
 
-      - uses: awalsh128/cache-apt-pkgs-action@v1.5.0
+      - name: "Run Lychee"
+        id: lychee
+        uses: lycheeverse/lychee-action@82202e5e9c2f4ef1a55a3d02563e1cb6041e5332
         with:
-          packages: libosmesa6-dev libgl1-mesa-dev python3-tk pandoc git-restore-mtime
-          version: 3.0
+          args: "--cache --no-progress --verbose './doc/source/**/*.md' './doc/source/**/*.rst' './doc/source/**/*.txt' './pyvista/**/*.py'"
```

Should probably also check examples:
```suggestion
          args: "--cache --no-progress --verbose './doc/source/**/*.md' './doc/source/**/*.rst' './doc/source/**/*.txt' './pyvista/**/*.py' './examples/**/*.py'"
```

---

+1